### PR TITLE
fix: query builder formula fails to eval

### LINKED ISSUE #1977: Dashboard: Formulae fails when multiple instance of same query used
Unable to calculate wanted to calculate `A/(A+B)` , but the same formulae altered to `1/(1+(B/A))` works.

Errors when same query is used multiple times in a single formulae.

_Error 1 for `A*A`:_

```
clickhouseReader/reader.go:2952	Error in processing query: code: 352, message: Ambiguous column 'A.value': While processing A.value
```

_Error 2 for `A/(A+B)`:_

```
clickhouseReader/reader.go:2952	Error in processing query: code: 62, message: Syntax error: failed at position 1243 ('GLOBAL'): GLOBAL INNER JOIN(SELECT k8s_node_name,  toStartOfInterval(toDateTime(intDiv(timestamp_ms, 1000)), INTERVAL 60 SECOND) as ts, avg(value) as value FROM signoz_me. Expected one of: FINAL, SAMPLE, USING, ON, end of query
```

_Screenshot:_

Screenshot from 2023-01-03 20-24-08

[Slack Message](https://signoz-team.slack.com/archives/C02B85R4R0A/p1672757744168279)

/cc @ankitnayan @srikanthccv

### PR DESCRIPTION
**Pull request by srikanthccv**
Fixes #1977 

When there are more than two variables involved in a formula, the query should be formed by specifying the `ON` clause. 

Earlier query built:
```sql
SELECT 
  A.service_name as service_name, 
  A.ts as ts, 
  A.value / (B.value + C.value) as value 
FROM (
    ...
) as A JOIN (
    ...
) as B JOIN(
    ...
) as C USING (service, ts)

```

Correct query:
```sql
SELECT 
  A.service_name as service_name, 
  A.ts as ts, 
  A.value / (B.value + C.value) as value 
FROM (
    ...
) as A JOIN (
    ...
) as B ON A.service_name = B.service_name AND A.ts = B.ts JOIN(
    ...
) as C ON B.service_name = C.service_name AND B.ts = C.ts

```

### CODE REVIEW

File: pkg/query-service/app/metrics/query_builder_test.go [hunk 1]
```diff
@@ -15,19 +16,19 @@ func TestBuildQuery(t *testing.T) {
 			Step:  60,
 			CompositeMetricQuery: &model.CompositeMetricQuery{
 				BuilderQueries: map[string]*model.MetricQuery{
-					"a": {
-						QueryName:         "a",
+					"A": {
```
Comment (srikanthccv) [inline pkg/query-service/app/metrics/query_builder_test.go:19]: The casing doesn't matter but updated to avoid confusion whit filter query values.

**Comment by ankitnayan** [comment]
Overall LGTM. We should add test cases to make sure existing queries are not broken when adding new capabilities

**Comment by prashant-shahi** [comment]
@srikanthccv failing to build frontend image in testingapp env. Tried from both workflow action and manual.

```
[3/5] Fetching packages...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
error An unexpected error occurred: "https://registry.yarnpkg.com/@ant-design/icons/-/icons-4.8.0.tgz: ESOCKETTIMEDOUT".
info If you think this is a bug, please open a bug report with the information provided in "/frontend/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
The command '/bin/sh -c CI=1 yarn install' returned a non-zero code: 1
make: *** [build-frontend-amd64] Error 1
```

**Comment by srikanthccv** [comment]
@prashant-shahi, that seems like a temporary availability issue of the package manger server. I couldn't reproduce it. Can you try again?

**Review by makeavish — APPROVED** [review]

**Pull request merged**